core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala

```diff
@@ -490,6 +490,40 @@ class PartKeyLuceneIndex(ref: DatasetRef,
     partKeySpan.finish()
     collector.result
   }
+
+  def partIdFromPartKeySlow(partKeyBase: Any,
+                            partKeyOffset: Long): Option[Int] = {
+
+    val columnFilters = schema.binSchema.toStringPairs(partKeyBase, partKeyOffset)
+      .map { pair => ColumnFilter(pair._1, Filter.Equals(pair._2)) }
+
+    val partKeySpan = Kamon.buildSpan("index-partition-lookup-latency")
+      .withTag("dataset", ref.dataset)
+      .withTag("shard", shardNum)
+      .start()
+    val booleanQuery = new BooleanQuery.Builder
+    columnFilters.foreach { filter =>
+      val q = leafFilter(filter.column, filter.filter)
+      booleanQuery.add(q, Occur.FILTER)
+    }
+    val query = booleanQuery.build()
+    logger.debug(s"Querying dataset=$ref shard=$shardNum partKeyIndex with: $query")
+    var chosenPartId: Option[Int] = None
+    def handleMatch(partId: Int, candidate: BytesRef): Unit = {
+      if (schema.binSchema.equals(partKeyBase, partKeyOffset,
```

Comment:
Not clear how this works

Reply:
ActionCollector accepts a function and invokes it for every matching result. Here we pass `handleMatch` as the function. It ensures that the partKey fully equals the match because a potential result could have additional tags. I'll add docs.